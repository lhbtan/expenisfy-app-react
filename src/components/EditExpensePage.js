import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense} from '../actions/expenses'

//refactor editexpense to be class based
// setup mapDispatch.. edit expense and remove expense

//should render editexpensepage
//snapshot

//should handle editexpense
//spies

//should handle removeExpense
//spies

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onClick = () => {
        this.props.removeExpense({id: this.props.expense.id});
        this.props.history.push('/');
    }
    render() {
        return  (
        <div>
        <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
        />
        <button 
        onClick={this.onClick}>Remove</button>
    </div>
        ); 
    } 
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);