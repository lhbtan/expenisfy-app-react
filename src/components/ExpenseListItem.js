import React from 'react';
import {BrowserRouter, Route, Switch, NavLink, Link} from 'react-router-dom';
import moment from 'moment';


const ExpenseListItem = ( { id, description, amount, createdAt }) => (
    <div>
    <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>
        Amount: {amount}, 
        Created At: {moment(createdAt).format('MMMM Do, YYYY')}
        </p>       
    </div>
);

export default ExpenseListItem;

