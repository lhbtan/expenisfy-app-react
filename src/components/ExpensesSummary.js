import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00') // use numeral here after fix lecture 140, 7:00
    return (
    <div>
    <p>Total of {expenseCount} {expenseWord}: {formattedExpenseTotal}</p>  
    </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary)