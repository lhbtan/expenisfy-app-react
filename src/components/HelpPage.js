import React from 'react';
import {BrowserRouter, Route, Switch, NavLink, Link} from 'react-router-dom';

const HelpPage = () => 
   (
    <div>
    HELP component.
   </div>
); 


export default HelpPage;