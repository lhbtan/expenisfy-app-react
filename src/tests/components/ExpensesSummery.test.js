import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly reder expenses summery with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly reder expenses summery with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={12} expensesTotal={123424233}/>);
    expect(wrapper).toMatchSnapshot();
})