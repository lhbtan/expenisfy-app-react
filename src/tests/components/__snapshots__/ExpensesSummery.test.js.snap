// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly reder expenses summery with 1 expense 1`] = `
<div>
  <p>
    Total of 
    1
     
    expense
    : 
    123
  </p>
</div>
`;

exports[`should correctly reder expenses summery with multiple expenses 1`] = `
<div>
  <p>
    Total of 
    12
     
    expenses
    : 
    123424233
  </p>
</div>
`;
